kernel/mouse/src/lib.rs

```diff
@@ -15,11 +15,51 @@ use ps2::{PS2Mouse, MousePacket};
 /// Because we perform the typical PIC remapping, the remapped IRQ vector number is 0x2C.
 const PS2_MOUSE_IRQ: u8 = interrupts::IRQ_BASE_OFFSET + 0xC;
 
+const PS2_MAX_MOUSE_BYTES: usize = 4;
+
 static MOUSE: Once<MouseInterruptParams> = Once::new();
 
+/// Everything we need in [`ps2_mouse_handler`].
 struct MouseInterruptParams {
     mouse: PS2Mouse<'static>,
     queue: Queue<Event>,
+    packet_bytes: PacketBytes,
+}
+
+/// Somewhat like an array/vec mixture to allow pushing single bytes
+/// of a mouse packet per interrupt into an array.
+/// This can handle MouseId 0 (3 bytes) and 3, 4 (4 bytes).
+struct PacketBytes {
+    len: usize,
```

Comment:
if `len` is always a statically-known value, we can use const generics here.

Reply:
Also `len` is a bit vague, give it a more precise name (and docs) that conveys it's an "expected byte count" or the "number of bytes in a complete mouse packet"